modules/harvest/tests/src/Functional/HarvestCodeConsolidationTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\datastore\Functional;
+
+use Drupal\FunctionalTests\Update\UpdatePathTestBase;
+use Drupal\harvest\HarvestService;
+use Drupal\sample_content\SampleContentService;
+use PHPUnit\Exception;
+
+/**
+ * Tests update functions for the datastore module.
+ *
+ * @group datastore
+ * @group update
+ */
+class HarvestCodeConsolidationTest extends UpdatePathTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setDatabaseDumpFiles() {
+    $this->databaseDumpFiles = [
+      __DIR__ . '/../../../../../tests/fixtures/update/update-2.20.0.php.gz',
+    ];
+  }
+
+  /**
+   * Test datastore module update 10001.
+   */
+  public function testUpdates10001on(): void {
+
+    // Set up sample content.
+    /** @var SampleContentService $sample_service */
+//    $sample_service = \Drupal::service('dkan.sample_content.service');
+//    $sample_service->registerSampleContentHarvest('sample_content');
+
+
+
+    // Get a baseline for pre-10001.
+    /** @var HarvestService $harvest_service */
+    $harvest_service = \Drupal::service('dkan.harvest.service');
+    try {
+      $result = $harvest_service->runHarvest('sample_content');
+    }
+    catch (\Exception $exception) {
+      $result = $exception->getMessage();
+    }
+    $this->assertSame('No items found to extract, review your harvest plan.', $result);
+
+    // Run update and recreate harvest service with updated data.
+    $this->runUpdates();
+    $harvest_service = \Drupal::service('dkan.harvest.service');
+
+
+    try {
+      $result = $harvest_service->runHarvest('sample_content');
+    }
+    catch (\Exception $exception) {
+      $result = $exception->getMessage();
+    }
+    $this->assertIsArray($result);
+    $this->assertSame('SUCCESS', $result['status']['extract']);
```

Comment:
@beeyayjay it's confusing to have try/catches here, because it's not clear when you read the test which one is happening. Is it throwing an exception in each case? If so, let's not set $result in the try, because it's not clear which one is in the assertion. If it's not throwing an exception, we don't need a try/catch at all.

Also not sure what this test shows, maybe I'm missing something. Why does the harvest have different results before and after the update? Aren't we trying to demonstrate that it works the same?

Reply:
@dafeder we're trying to demonstrate that the update works as intended. `runHarvest()` fails before the update because the extractor namespace in the db is outdated. It should succeed after the update is applied. (Note that the second assignment of `$harvest_service` on L54 is required to load the updated data.)

I think I can replace the first try/catch with an `expectException()` and leave the second catch blank. The second `runHarvest()` *shouldn't* fail, but that's really what we're testing, and a failed assertion seems preferable to an uncaught exception if the test fails.

Reply:
I had to leave the first try/catch block in. `expectException()` worked, but the rest of the test didn't run.

I rewrote it a bit though, so hopefully it's clearer now.

Reply:
OK sorry re-reading this the logic of the test does make sense. But I'm still failing to see why the second try/catch is necessary. Removing the `$result =` does make the first more readable and the test more valid, thanks. See my suggestion below.